integration-tests/test_registration.py

```diff
@@ -93,3 +98,38 @@ def test_register_group_option(insights_client, legacy_upload_value):
         check=False,
     )
     assert register_group_option.returncode == 0
+
+def test_machineid_changes_when_subman_is_unregistered_and_registered(insights_client, subman, test_config):
+    """machine-id should be different based on the auth method"""
+    if 'satellite' in test_config.environment:
+        pytest.skip("this test requires BASIC authentication")
+
+    insights_client.unregister()
```

Comment:
insights_client.unregister() will fail at this point if already unregistered, you can try adding insights_client.unregister(check=False)

Reply:
strictly speaking you don't need to explicitly `unregister()`, as using the `insights_client` fixture (as used elsewhere) ensures that `insights-client` is unregistered automatically at the end of each test

Reply:
Like Pino said we don't need to unregister so it can be removed also.